### ISSUE TITLE
Maps: Does not support overriding accessibility

### ISSUE THREAD

**Issue by jacob-williams-kr** [googlemaps/android-maps-compose#305]
#### Environment details

Compose Maps: `com.google.maps.android:maps-compose:2.11.4`

#### Steps to reproduce

The Compose `GoogleMap` function does not support overriding the accessibility options. In my case, I am trying to get the map to not be read out by screen readers. Previously, with the XML setup, we did this with `android:importantForAccessibility="noHideDescendants"`, however using `clearAndSetSemantics` does not work for the compose view.

#### Code example

```
GoogleMap(
        cameraPositionState = cameraPositionState,
        modifier = Modifier.clearAndSetSemantics {},
) 
```

Markers also do not support a custom modifier, meaning they cannot have custom screen readouts, or custom tagging for tests

**Comment by wangela** [comment]
If you would like to upvote the priority of this issue, please comment below or react with :+1: so we can see what is popular when we triage.
--------

@jacob-williams-kr Thank you for opening this issue. 🙏
Please check out these other resources that might help you get to a resolution in the meantime:

- [Check the issue tracker](https://developers.google.com/maps/support/#issue_tracker) - bugs and feature requests for Google Maps Platform APIs and SDKs
- [Open a support case](https://developers.google.com/maps/support#contact-maps-support) - Get 1:1 support in Cloud Console.
- [Discord](https://discord.gg/fHTDmZe9En) - chat with other developers
- [StackOverflow](http://stackoverflow.com/questions/tagged/google-maps) - use the `google-maps` tag


_This is an automated message, feel free to ignore._

**Comment by lucas-livefront** [comment]
This is significantly affecting me. I am making new features for an app with millions of users.

**Comment by lucas-livefront** [comment]
@wangela & @jacob-williams-kr I apologize for @ing you but I am getting a little desperate. I will  need to make a talk back work around for this issue so I just want to check in for an update here before I start some complex solution.

I did make this [SO post](https://stackoverflow.com/questions/76282872/ignore-for-accessibility-with-google-maps-compose-sdk) as well to no avail.

**Comment by BigAdam22** [comment]
This issue significantly impacts our project's efforts to enhance accessibility support.

We cannot modify or remove semantic properties using modifiers.

Moreover, TalkBack consistently announces "Double tap to activate," regardless of whether we include any onClick callbacks. This behavior remains even when we're exclusively utilizing it as a non-interactive static map, without any enabled gestures or controls.

It's worth noting that tapping directly on a marker also triggers focus when using TalkBack. It will be announced as "_Marker title_ Double tap to activate."

To mitigate this, the only solution we've found for the map is as follows:

```kotlin
Box(
    modifier = modifier
        .semantics(mergeDescendants = true) { }
        .clearAndSetSemantics { contentDescription = "Google Map" }
) {
    GoogleMap(
         ... 
```

This approach ensures that the map itself is not announced as an interactive element. However, this workaround cannot be applied to markers.

**Comment by mightymia** [comment]
We have a large app that services millions of people worldwide every day; this is a significant issue for our travelers using talkback. I want the option to remove the click on the map and hide the marker from talkback because, in our use case, it is for decoration purposes only.

**Comment by wangela** [comment]
Thank you for sharing how this can help wtih accessibility for apps built with the Maps Compose library. Raising the priority and will evaluate whether we can raise it even further.

**Comment by wangela** [comment]
Inviting the folks here who have mentioned markers to review PR #407 and request modifications to provide the necessary accessibility support for markers.

**Comment by mightymia** [comment]
This is great, thank you! When can we expect this to be available to use?

**Comment by mightymia** [comment]
Thank you @kikoso, I've made a note in our ticket to update to the 4.2.0 version for the fix.

**Comment by lucas-livefront** [comment]
Bravo.

**Comment by esalavastru** [comment]
By any chance, is there any way of overriding the `contentDescription` text for a `Clustering`'s  `clusterContent` composable or `clusterItemContent` composable?

Currently I'm adding `contentDescription` to the custom composables for the Clustering, and TalkBack is just saying "MapMarker" instead of the String that I added.
